nextcord/client.py

```diff
@@ -1909,54 +1965,236 @@ def add_application_command(
+            self,
+            data: Optional[Dict[Optional[int], List[dict]]] = None,
+            use_rollout: bool = True,
+            associate_known: bool = True,
+            delete_unknown: bool = True,
+            update_known: bool = True,
+            register_new: bool = True,
+    ) -> None:
+        """|coro|
+
+        Syncs all application commands with Discord. Will sync global commands if any commands added are global, and
+        syncs with all guilds that have an application command targeting them.
+
+        This may call Discord many times depending on how different guilds you have local commands for, and how many
+        commands Discord needs to be updated or added, which may cause your bot to be rate limited or even Cloudflare
+        banned in VERY extreme cases.
+
+        This may incur high CPU usage depending on how many commands you have and how complex they are, which may cause
+        your bot to halt while it checks local commands against the existing commands that Discord has.
+
+        For a more targeted version of this method, see :func:`Client.sync_application_commands`
+
+        Parameters
+        ----------
+        data: Optional[Dict[Optional[:class:`int`], List[:class:`dict`]]]
+            Data to use when comparing local application commands to what Discord has. The key should be the
+            :class:`int` guild ID (`None` for global) corresponding to the value list of application command payloads
+            from Discord. Any guild ID's not provided will be fetched if needed. Defaults to `None`
+        use_rollout: :class:`bool`
+            If the rollout guild IDs of commands should be used. Defaults to `True`
+        associate_known: :class:`bool`
+            If local commands that match a command already on Discord should be associated with each other.
+            Defaults to `True`
+        delete_unknown: :class:`bool`
+            If commands on Discord that don't match a local command should be deleted. Defaults to `True`
+        update_known: :class:`bool`
+            If commands on Discord have a basic match with a local command, but don't fully match, should be updated.
+            Defaults to `True`
+        register_new: :class:`bool`
+            If a local command that doesn't have a basic match on Discord should be added to Discord.
+            Defaults to `True`
+        """
+        # All this does is passthrough to connection state. All documentation updates should also be updated
+        # there, and vice versa.
+        await self._connection.sync_all_application_commands(
+            data=data, use_rollout=use_rollout, associate_known=associate_known, delete_unknown=delete_unknown,
+            update_known=update_known, register_new=register_new
+        )
+
+    async def sync_application_commands(
             self,
             data: Optional[List[dict]] = None,
+            guild_id: Optional[int] = None,
+            associate_known: bool = True,
```

Comment:
Likely make this onwards kwargs only so people dont do the wrong arg thinking its something else?